composer/datasets/ade20k.py

```diff
@@ -388,6 +391,155 @@ def initialize_object(self, batch_size, dataloader_hparams) -> DataSpec:
                         device_transforms=device_transform_fn)
 
 
+class StreamingADE20k(StreamingDataset):
+    """
+    Implementation of the ADE20k dataset using StreamingDataset.
+
+    Args:
+        remote (str): Remote directory (S3 or local filesystem) where dataset is stored.
+        local (str): Local filesystem directory where dataset is cached during operation.
+        split (str): The dataset split to use, either 'train', or 'val'. Default: ``'train```.
+        shuffle (bool): Whether to shuffle the samples in this dataset.
+        base_size (int): initial size of the image and target before other augmentations. Default: ``512``.
+        min_resize_scale (float): the minimum value the samples can be rescaled. Default: ``0.5``.
+        max_resize_scale (float): the maximum value the samples can be rescaled. Default: ``2.0``.
+        final_size (int): the final size of the image and target. Default: ``512``.
+    """
+
+    def decode_uid(self, data: bytes) -> Any:
+        return data.decode('utf-8')
+
+    def decode_image(self, data: bytes) -> Any:
+        return Image.open(BytesIO(data))
+
+    def decode_annotation(self, data: bytes) -> Any:
+        return Image.open(BytesIO(data))
+
+    def __init__(self,
+                 remote: str,
+                 local: str,
+                 split: str,
+                 shuffle: bool,
+                 base_size: int = 512,
+                 min_resize_scale: float = 0.5,
```

Comment:
Should the dataset class be opinionated about the transformations? (Realize you likely made this change given the previous feedback on the PR!) If the answer is no, then OK to leave this as is and clean up later -- potentially once torch data pipes come out.

Reply:
I think we want to stay opinionated (and yes that was based on comments earlier in the PR :), as I think our goal is not to build an ADE20k for someone to re-use (other than basic functionality), rather, ADE20k is an example that a user will look at in order to build their own custom datasets, with custom processing.